src/app/services/xml-parser-service/xmp-parser.service.ts

```diff
@@ -0,0 +1,179 @@
+export type Paragraph = org.cru.godtools.shared.tool.parser.model.Paragraph;
+export type Resource = org.cru.godtools.shared.tool.parser.model.Resource;
+export type Manifest = org.cru.godtools.shared.tool.parser.model.Manifest;
+export type XmlParserData =
+  org.cru.godtools.shared.tool.parser.ParserResult.Data;
+export type Video = org.cru.godtools.shared.tool.parser.model.Video;
+export type Button = org.cru.godtools.shared.tool.parser.model.Button;
+export type EventId = org.cru.godtools.shared.tool.parser.model.EventId;
+export type Form = org.cru.godtools.shared.tool.parser.model.Form;
+export type Input = org.cru.godtools.shared.tool.parser.model.Input;
+export type Link = org.cru.godtools.shared.tool.parser.model.Link;
+
+export type Spacer = org.cru.godtools.shared.tool.parser.model.Spacer;
+export type Tabs = org.cru.godtools.shared.tool.parser.model.Tabs;
+export type Tab = org.cru.godtools.shared.tool.parser.model.Tabs.Tab;
+export type Accordion = org.cru.godtools.shared.tool.parser.model.Accordion;
+export type ModalCard = org.cru.godtools.shared.tool.parser.model.Card;
+export type Multiselect = org.cru.godtools.shared.tool.parser.model.Multiselect;
+
+export namespace XmlParser {
+  export const ParserConfig = org.cru.godtools.shared.tool.parser.ParserConfig;
+  export const ManifestParser = manifestParser;
+}
+
+export const ContentParser = (content: any): string => {
+  if (content instanceof org.cru.godtools.shared.tool.parser.model.Image) {
+    console.log('CONTENT: Image');
+    return 'image';
+  } else if (
+    content instanceof org.cru.godtools.shared.tool.parser.model.Text
+  ) {
+    console.log('CONTENT: Text');
+    return 'text';
+  } else if (
+    content instanceof org.cru.godtools.shared.tool.parser.model.Paragraph
+  ) {
+    console.log('CONTENT: Paragraph');
+    return 'paragraph';
+  } else if (
+    content instanceof org.cru.godtools.shared.tool.parser.model.Video
+  ) {
+    console.log('CONTENT: Video');
+    return 'video';
+  } else if (
+    content instanceof org.cru.godtools.shared.tool.parser.model.Button
+  ) {
+    console.log('CONTENT: Button');
+    return 'button';
+  } else if (
+    content instanceof org.cru.godtools.shared.tool.parser.model.Form
+  ) {
+    console.log('CONTENT: Form');
+    return 'form';
+  } else if (
+    content instanceof org.cru.godtools.shared.tool.parser.model.Input
+  ) {
+    console.log('CONTENT: Input');
+    return 'input';
+  } else if (
+    content instanceof org.cru.godtools.shared.tool.parser.model.Spacer
+  ) {
+    console.log('CONTENT: Spacer');
+    return 'spacer';
+  } else if (
+    content instanceof org.cru.godtools.shared.tool.parser.model.Link
+  ) {
+    console.log('CONTENT: Link');
+    return 'link';
+  } else if (
+    content instanceof org.cru.godtools.shared.tool.parser.model.Tabs
+  ) {
+    console.log('CONTENT: Tabs');
+    return 'tabs';
+  } else if (
+    content instanceof org.cru.godtools.shared.tool.parser.model.Tabs.Tab
```

Comment:
I don't believe this case will ever return true, `Tab` doesn't (shouldn't) extend the `Content` type, which means it should never be an element in a `Content` array.

Reply:
You're right; we don't have a `tab` content type. This must have been left in after testing and building the tab component. I've removed it.